### ISSUE TITLE
 sdl1[core]:x86-windows-static build failure

### ISSUE THREAD

**Issue by gdiObjects** [microsoft/vcpkg#8281]
Windows 10, Visual Studio 2017

**To Reproduce**
`./vcpkg upgrade --no-dry-run`

**Failure logs** 
Starting package 36/48: sdl1:x86-windows-static
Building package sdl1[core]:x86-windows-static...
-- Using cached D:/Development/Projects/vcpkg/downloads/SDL-Mirror-SDL-release-1.2.15.tar.gz
-- Using source at D:/Development/Projects/vcpkg/buildtrees/sdl1/src/ase-1.2.15-086d8ac6c2
-- Up-to-date: D:/Development/Projects/vcpkg/buildtrees/sdl1/src/ase-1.2.15-086d8ac6c2/VisualC/SDL/SDL.vcxproj
-- Up-to-date: D:/Development/Projects/vcpkg/buildtrees/sdl1/src/ase-1.2.15-086d8ac6c2/VisualC/SDLmain/SDLmain.vcxproj
-- Building VisualC/SDL1_2017.sln for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: msbuild D:/Development/Projects/vcpkg/buildtrees/sdl1/x86-windows-static-rel/ase-1.2.15-086d8ac6c2/VisualC/SDL1_2017.sln /p:Configuration=Release /t:Rebuild /p:Platform=Win32 /p:PlatformToolset=v141 /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /p:WindowsTargetPlatformVersion=10.0.18362.0 /m /p:WholeProgramOptimization=false
    Working Directory: D:/Development/Projects/vcpkg/buildtrees/sdl1/x86-windows-static-rel/ase-1.2.15-086d8ac6c2
    Error code: 1
    See logs for more information:
      D:\Development\Projects\vcpkg\buildtrees\sdl1\build-x86-windows-static-rel-out.log
[build-x86-windows-static-rel-out.log](https://github.com/microsoft/vcpkg/files/3638766/build-x86-windows-static-rel-out.log)

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_msbuild.cmake:157 (vcpkg_execute_required_process)
  ports/sdl1/portfile.cmake:31 (vcpkg_install_msbuild)
  scripts/ports.cmake:91 (include)

**Comment by JackBoosY** [comment]
I can repro it. Even #8327 merged.

**Comment by JackBoosY** [comment]
```
-- Performing post-build validation
DLLs should not be present in a static build, but the following DLLs were found:

    G:/10082/vcpkg/packages/sdl1_x64-windows-static/bin/SDL.dll
    G:/10082/vcpkg/packages/sdl1_x64-windows-static/debug/bin/SDL.dll

There should be no bin\ directory in a static build, but G:\10082\vcpkg\packages\sdl1_x64-windows-static\bin is present.
There should be no debug\bin\ directory in a static build, but G:\10082\vcpkg\packages\sdl1_x64-windows-static\debug\bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()

Expected Debug,Static crt linkage, but the following libs had invalid crt linkage:

    G:/10082/vcpkg/packages/sdl1_x64-windows-static/debug/lib/manual-link/SDLmaind.lib: Release,Dynamic

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Expected Release,Static crt linkage, but the following libs had invalid crt linkage:

    G:/10082/vcpkg/packages/sdl1_x64-windows-static/lib/manual-link/SDLmain.lib: Release,Dynamic

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Found 4 error(s). Please correct the portfile:
    G:\10082\vcpkg\ports\sdl1\portfile.cmake
-- Performing post-build validation done
Error: Building package sdl1:x64-windows-static failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: sdl1:x64-windows-static
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.
```

**Comment by JackBoosY** [comment]
Hi @gdiObjects, thanks for reporting this issue.
I'v fixed it in #10105, please update vcpkg and rebuild sdl1.

Thanks.
